src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseCaseReport.java

```diff
@@ -169,7 +169,8 @@ protected void collectData(PineryClient pinery) throws HttpResponseException, IO
                                 || "POS"
                                     .equals(
                                         getUpstreamAttribute(
-                                            ATTR_GROUP_ID, s, allPreciseSamplesById)))
+                                            ATTR_GROUP_ID, s, allPreciseSamplesById))
+                                || null == getAttribute(ATTR_GROUP_ID, s))
```

Comment:
This is only checking upstream (parent samples) for "POS" and only checking the current sample for null? Probably want to check both in both cases?

Reply:
Ilinca says the current counts are correct, and adding the upstream null check does change the count :shrug:

Reply:
(It does also check the current sample for POS too though

Reply:
Ah, I missed the other check for "POS" on current sample.

`getUpstreamAttribute` should be checked for null too though. There could be a tissue with group ID "NEG" with a child slide not specifying a group ID. In this case, the tissue's "NEG" group ID should apply to the child, but the report will count it as "POS" instead. While there may not be any samples matching this pattern right now, there could be in the future before this report is rerun.

Reply:
sorry, I'd misunderstood how this check should be implemented. Fixed and verified no data change